src/compile/data/formatparse.ts

```diff
@@ -25,38 +26,45 @@ export namespace formatParse {
           return;
         }
         parseComponent[fieldDef.field] = 'number';
+      } else {
+        // the field should not be parsed
+        parseComponent[fieldDef.field] = null;
       }
     });
     return parseComponent;
   }
 
-  export const parseUnit = parse;
-
-  export function parseFacet(model: FacetModel) {
-    let parseComponent = parse(model);
-
-    // If child doesn't have its own data source, but has its own parse, then merge
-    const childDataComponent = model.child().component.data;
-    if (!childDataComponent.source && childDataComponent.formatParse) {
-      extend(parseComponent, childDataComponent.formatParse);
-      delete childDataComponent.formatParse;
-    }
-    return parseComponent;
+  /**
+   * Merge the format parse from the parent and all components into parent.
+   *
+   * Format parse is only merged up if the parent either does not define a parse or the same parse.
+   */
+  export function merge(dataComponent: DataComponent, childDataComponents: DataComponent[]) {
+    childDataComponents.reduce((collector, childData) => {
+      if (childData.formatParse) {
+        forEach(childData.formatParse, (parse, field) => {
+          if (parse === collector[field] || collector[field] === undefined) {
+            collector[field] = parse;
+            delete childData.formatParse[field];
+          }
+        });
+      }
+      return collector;
+    }, dataComponent.formatParse);
```

Comment:
This only returns the merge results, but does not return whether the merge completes.  

Don't you have to care if the something cannot be merged, then terminate the overall merge process?